libstat/NetworkStatHandler.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2020 fisco-dev contributors.
+ */
+/**
+ * @brief : Implement network statistics
+ * @file: NetworkStatHandler.cpp
+ * @author: yujiechen
+ * @date: 2020-03-20
+ */
+#include "NetworkStatHandler.h"
+
+using namespace dev;
+using namespace dev::stat;
+// update incoming traffic
+void NetworkStatHandler::updateIncomingTraffic(int32_t const& _msgType, uint64_t _msgSize)
+{
+    m_totalInMsgBytes += _msgSize;
+    WriteGuard l(x_InMsgTypeToBytes);
+    updateTraffic(m_InMsgTypeToBytes, _msgType, _msgSize);
+}
+
+// update outcoming traffic
+void NetworkStatHandler::updateOutcomingTraffic(int32_t const& _msgType, uint64_t _msgSize)
+{
+    m_totalOutMsgBytes += _msgSize;
+    WriteGuard l(x_OutMsgTypeToBytes);
+    updateTraffic(m_OutMsgTypeToBytes, _msgType, _msgSize);
+}
+
+void NetworkStatHandler::updateTraffic(StatPtr _statMap, int32_t const& _msgType, uint64_t _msgSize)
+{
+    if (_statMap->count(_msgType))
+    {
+        (*_statMap)[_msgType] += _msgSize;
+        return;
+    }
+    (*_statMap)[_msgType] = _msgSize;
+}
+
+void NetworkStatHandler::printStatistics()
+{
+    printStatistics(c_p2p_statisticName, m_p2pMsgTypeToDesc, m_p2pMsgTypeToDesc);
+    printStatistics(c_sdk_statisticName, c_sdkInMsgTypeToDesc, c_sdkOutMsgTypeToDesc);
+    // print the totalBytes
+    STAT_LOG(INFO) << LOG_TYPE("Total") << LOG_KV("g", m_groupId)
+                   << LOG_KV("Total_In", m_totalInMsgBytes)
+                   << LOG_KV("Total_Out", m_totalOutMsgBytes);
+}
+
+void NetworkStatHandler::printStatistics(std::string const& _statName,
+    std::map<int32_t, std::string> _inStatTypeToDesc,

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter '_inStatTypeToDesc' should be passed by const reference.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5223061)